source/include/storage/cs_IpcRamBluenet.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Author: Crownstone Team
+ * Copyright: Crownstone (https://crownstone.rocks)
+ * Date: Oct 24, 2022
+ * License: LGPLv3+, Apache License 2.0, and/or MIT (triple-licensed)
+ */
+
+#pragma once
+
+#include <ipc/cs_IpcRamDataContents.h>
+
+/**
+ * Class to operate on bluenet to bluenet IPC ram.
+ */
+class IpcRamBluenet {
+public:
+	/**
+	 * Get the singleton instance.
+	 */
+	static IpcRamBluenet& getInstance();
+
+	/**
+	 * Reads the IPC ram.
+	 */
+	void init();
+
+	/**
+	 * Whether the IPC ram data was valid on boot.
+	 */
+	bool isValidOnBoot();
+
+	/**
+	 * Get the current data.
+	 *
+	 * Don't forget to first check whether the data is valid on boot.
+	 */
+	const bluenet_ipc_bluenet_data_t& getData();
+
+	//! Update the energy field.
+	void updateEnergyUsed(const int64_t& energyUsed);
+
+	//! Update a microapp field.
+	void updateMicroappData(uint8_t appIndex, const microapp_reboot_data_t& data);
+
+private:
+	//! Constructor, singleton, thus made private
+	IpcRamBluenet();
+
+	//! Copy constructor, singleton, thus made private
+	IpcRamBluenet(IpcRamBluenet const&)            = delete;
+
+	//! Assignment operator, singleton, thus made private
+	IpcRamBluenet& operator=(IpcRamBluenet const&) = delete;
+
+	//! Cache the data in IPC.
+	bluenet_ipc_data_payload_t _ipcData;
```

Comment:
considering memory usage: 
This cache variable is only used when writing. Doesn't it make sense to just read the previous contents of ram before erase-update? Copying is cheap in ram anyway.

Reply:
Ah, you return data by reference in `getData();` That would have to change to a copy as well in that case. Are we reading from ipc a lot?

Reply:
At the moment, reading is only done on boot really. Writing is done every 20ms.